### ISSUE TITLE
Fix tab ordering (in Tab Tray) behaviour to new spec

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#9593]
New tab ordering behaviour should be respected - see Acceptance Criteria

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3493)

**Comment by data-sync-user** [comment]
➤ Simion Basca commented:

Verified fix on main 627116c214d9d74bb5a391f59999cd4eb1f68eee
